File: Examples/Objective-C/Remote Control/Remote Control/ImagePickerViewController.m
//
//  ImageViewController.m
//  Remote Control
//
//  Created by Calvin on 7/7/17.
//  Copyright © 2017 Bridgefy Inc. All rights reserved.
//

#import "ImagePickerViewController.h"
#import "AdminViewTabBarController.h"

@interface ImagePickerViewController ()

@property (nonatomic, retain) UIButton *currentButton;

@end

@implementation ImagePickerViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (IBAction)imageButtonPressed:(id)sender {
    UIButton *selectedButton = (UIButton *)sender;
    
    [self.currentButton setImage:[UIImage imageNamed:@"radioButtonOff"]
                        forState:UIControlStateNormal];
    
    [selectedButton setImage:[UIImage imageNamed:@"radioButtonOn"]
                    forState:UIControlStateNormal];
    
    self.currentButton = selectedButton;
    
    if (self.tabBarController) {
        AdminViewTabBarController *tabBarController = (AdminViewTabBarController *)self.tabBarController;
        [tabBarController sendObject:@(selectedButton.tag) sender:self];
    }
}


@end

File: Examples/Swift/Remote Control/Remote Control/ImagePickerViewController.swift
//
//  ImagePickerViewController.swift
//  Remote Control
//
//  Created by Calvin on 7/11/17.
//  Copyright © 2017 Bridgefy Inc. All rights reserved.
//

import UIKit

class ImagePickerViewController: UIViewController {
    
    var currentButton: UIButton?
    
    override func viewDidLoad() {
        super.viewDidLoad()

        // Do any additional setup after loading the view.
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    @IBAction func imageButtonPressed(_ sender: Any) {
        let selectedButton = sender as! UIButton
        
        self.currentButton?.setImage(#imageLiteral(resourceName: "radioButtonOff"), for: .normal)
        selectedButton.setImage(#imageLiteral(resourceName: "radioButtonOn"), for: .normal)
        self.currentButton = selectedButton
        
        if (self.tabBarController != nil) {
            let tabBarController = self.tabBarController as! AdminViewTabBarController
            tabBarController.send(object: selectedButton.tag as Any, sender: self)
        }
    }

}
